pkg/services/secrets/kvstore/migrations/datasources_mig.go

```diff
@@ -0,0 +1,85 @@
+package migrations
+
+import (
+	"context"
+	"encoding/json"
+
+	"github.com/grafana/grafana/pkg/bus"
+	"github.com/grafana/grafana/pkg/infra/log"
+	"github.com/grafana/grafana/pkg/services/datasources"
+	"github.com/grafana/grafana/pkg/services/featuremgmt"
+	"github.com/grafana/grafana/pkg/services/secrets/kvstore"
+	"github.com/grafana/grafana/pkg/services/sqlstore"
+	"github.com/grafana/grafana/pkg/setting"
+)
+
+const (
+	dataSourceSecretType = "datasource"
+)
+
+type DataSourceSecretMigrationService struct {
+	sqlStore           *sqlstore.SQLStore
+	dataSourcesService datasources.DataSourceService
+	secretsStore       kvstore.SecretsKVStore
+	features           featuremgmt.FeatureToggles
+	log                log.Logger
+	bus                bus.Bus
+}
+
+func ProvideDataSourceMigrationService(
+	cfg *setting.Cfg, dataSourcesService datasources.DataSourceService,
+	secretsStore kvstore.SecretsKVStore, features featuremgmt.FeatureToggles,
+	sqlStore *sqlstore.SQLStore, bus bus.Bus,
+) *DataSourceSecretMigrationService {
+	return &DataSourceSecretMigrationService{
+		sqlStore:           sqlStore,
+		dataSourcesService: dataSourcesService,
+		secretsStore:       secretsStore,
+		features:           features,
+		log:                logger,
+		bus:                bus,
+	}
+}
+
+func (s *DataSourceSecretMigrationService) Run(ctx context.Context) error {
+	return s.sqlStore.InTransaction(ctx, func(ctx context.Context) error {
+		query := &datasources.GetDataSourcesQuery{}
+		err := s.dataSourcesService.GetDataSources(ctx, query)
+		if err != nil {
+			return err
+		}
+		for _, ds := range query.Result {
+			hasMigration, _ := ds.JsonData.Get("secretMigrationComplete").Bool()
```

Comment:
Use `MustBool` or checking the error

Reply:
Will do!

Reply:
Done on https://github.com/grafana/grafana/pull/50463/commits/b153ef1b0d7f75c9cb9199842523eaa1341b9bce